snuba/snuba_migrations/spans_v2/0001_spans_v2.py

```diff
@@ -0,0 +1,189 @@
+        )
+        for i in range(50)
+    ]
+    + [
+        Column(
+            f"attr_num_{i}",
+            Map(String(), Float(64), modifiers=Modifiers(codecs=["ZSTD(1)"])),
+        )
+        for i in range(50)
+    ]
+    + [
+        Column(
+            f"attr_bool_{i}",
+            Map(String(), Bool(), modifiers=Modifiers(codecs=["ZSTD(1)"])),
+        )
+        for i in range(10)
+    ]
+)
+
+index_create_ops: Sequence[SqlOperation] = (
+    [
+        operations.AddIndex(
+            storage_set=storage_set_name,
+            table_name=local_table_name,
+            index_name="bf_trace_id",
+            index_expression="trace_id",
+            index_type="bloom_filter",
+            granularity=1,
+            target=OperationTarget.LOCAL,
+        ),
+    ]
+    + [
+        operations.AddIndex(
+            storage_set=storage_set_name,
+            table_name=local_table_name,
+            index_name=f"bf_attr_str_{i}",
+            index_expression=f"mapKeys(attr_str_{i})",
+            index_type="bloom_filter",
+            granularity=1,
+            target=OperationTarget.LOCAL,
+        )
+        for i in range(50)
+    ]
+    + [
+        operations.AddIndex(
+            storage_set=storage_set_name,
+            table_name=local_table_name,
+            index_name=f"bf_attr_str_val_{i}",
+            index_expression=f"mapValues(attr_str_{i})",
+            index_type="ngrambf_v1(4, 1024, 10, 1)",
+            granularity=1,
+            target=OperationTarget.LOCAL,
+        )
+        for i in range(50)
+    ]
+    + [
+        operations.AddIndex(
+            storage_set=storage_set_name,
+            table_name=local_table_name,
+            index_name=f"bf_attr_num_{i}",
+            index_expression=f"mapKeys(attr_num_{i})",
+            index_type="bloom_filter",
+            granularity=1,
+            target=OperationTarget.LOCAL,
+        )
+        for i in range(50)
+    ]
+    + [
+        operations.AddIndex(
+            storage_set=storage_set_name,
+            table_name=local_table_name,
+            index_name=f"bf_attr_bool_{i}",
+            index_expression=f"mapKeys(attr_bool_{i})",
+            index_type="bloom_filter",
+            granularity=1,
+            target=OperationTarget.LOCAL,
+        )
+        for i in range(10)
+    ]
+)
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_ops(self) -> Sequence[SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=storage_set_name,
+                table_name=local_table_name,
+                columns=columns,
+                engine=table_engines.CollapsingMergeTree(
+                    primary_key="(organization_id, _sort_timestamp, trace_id)",
+                    order_by="(organization_id, _sort_timestamp, trace_id, span_id)",
+                    sign_column="sign",
+                    partition_by="(toMonday(_sort_timestamp))",
```

Comment:
I'd partition by day instead of week, ClickHouse will still be happy and this might prove useful for the mutability case (cc @untitaker )

Reply:
there's definitely downsides to producing this many partitions. I think at this point the decision is a bit arbitrary as I believe we may have to revisit it again anyway with very high likelihood.